mmv1/third_party/terraform/services/datastream/resource_datastream_connection_profile_test.go

```diff
@@ -247,3 +272,88 @@ resource "google_datastream_connection_profile" "mysql_con_profile" {
 }
 `, context)
 }
+
+func testAccDatastreamConnectionProfile_SSHKey_Update(context map[string]interface{}, preventDestroy bool, private_key string) string {
+	context["lifecycle_block"] = ""
+	if preventDestroy {
+		context["lifecycle_block"] = `
+        lifecycle {
+            prevent_destroy = true
+        }`
+	}
+
+	context["private_key"] = private_key
+
+	return acctest.Nprintf(`
+resource "google_sql_database_instance" "instance" {
+        name             = "tf-test-my-database-instance%{random_suffix}"
+        database_version = "POSTGRES_14"
+        region           = "us-central1"
+        settings {
+            tier = "db-f1-micro"
+            ip_configuration {
+    
+            // Datastream IPs will vary by region.
+            authorized_networks {
+                value = "34.71.242.81"
+            }
+    
+            authorized_networks {
+                value = "34.72.28.29"
+            }
+    
+            authorized_networks {
+                value = "34.67.6.157"
+            }
+    
+            authorized_networks {
+                value = "34.67.234.134"
+            }
+    
+            authorized_networks {
+                value = "34.72.239.218"
+            }
+        }
+    }
+    
+        deletion_protection  = "false"
+    }
+    
+    resource "google_sql_database" "db" {
+        instance = google_sql_database_instance.instance.name
+        name     = "db"
+    }
+    
+    resource "random_password" "pwd" {
+        length = 16
+        special = false
+    }
+    
+    resource "google_sql_user" "user" {
+        name = "user"
+        instance = google_sql_database_instance.instance.name
+        password = random_password.pwd.result
+    }
+
+resource "google_datastream_connection_profile" "ssh_connectivity_profile" {
+    display_name          = "Source connection profile"
+    location              = "us-central1"
+    connection_profile_id = "tf-test-mysql-profile%{random_suffix}"
+
+    postgresql_profile {
+        hostname = google_sql_database_instance.instance.public_ip_address
+        username = google_sql_user.user.name
+        password = google_sql_user.user.password
+        database = google_sql_database.db.name
+    }
+
+    forward_ssh_connectivity {
+        hostname = google_sql_database_instance.instance.public_ip_address
+        username = google_sql_user.user.name
+        port     = 22
+        private_key = "%{private_key}"
```

Comment:
I'm no real familiar with the use case, but just in case it helps, this test only needs to do enough to use the private_key. If the other config here is extraneous, it could be removed to make things more focused, and possibly faster to deploy.